modules/openthread/platform/misc.c

```diff
@@ -9,6 +9,13 @@
 #include <openthread/instance.h>
 #include <openthread/platform/misc.h>
 
+#if defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_RETENTION)
+#include <zephyr/retention/bootmode.h>
+#endif
+#if defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_GPIO)
+#include <zephyr/drivers/gpio.h>
+#endif
```

Comment:
Nit: the options are in a `choice`, so prefer to use `#elif`

I think it makes sense to add a `BUILD_ASSERT` here if no gpio instance was set? Returning `OT_ERROR_NOT_CAPABLE` doesn't look very useful.

```c
#if defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_RETENTION)

#include <zephyr/retention/bootmode.h>

#elif defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_GPIO)

BUILD_ASSERT(DT_HAS_COMPAT_STATUS_OKAY(openthread_config), "...");
BUILD_ASSERT(DT_NODE_HAS_PROP(DT_COMPAT_GET_ANY_STATUS_OKAY(openthread_config), bootloader_gpios), "...");

#include <zephyr/drivers/gpio.h>

static const struct gpio_dt_spec bootloader_gpio =
	GPIO_DT_SPEC_GET(DT_COMPAT_GET_ANY_STATUS_OKAY(openthread_config),
			 bootloader_gpios);
#endif
```

Reply:
Seems more clear, thanks